### ISSUE TITLE
Bar Series Tooltip Format Y Value

### ISSUE THREAD

**Issue by walsha2** [syncfusion/flutter-examples#29]
### Description:

This is a re-open of #22 which was closed (I can't reopen issues because I am not a collaborator on the project), but this still is an issue. Here are more details and concrete example to reproduce:

```dart
tooltipBehavior: TooltipBehavior(
	enable: true,
	header: 'Tooltip Header',
	format: "point.x has value of: point.y",
),
```

<p align="center">
<img src="https://user-images.githubusercontent.com/53703092/65290118-6d87f900-db13-11e9-82d1-b07cb0fb5420.jpeg" width="400">
</p>

### Code To Reproduce: 

As you can see from the above, `point.y` does not get replaced. Please run this simple `main.dart` to reproduce. All the code lives within it, attached below as `txt` file:

[main.txt](https://github.com/syncfusion/flutter-examples/files/3633662/main.txt)

**Comment by walsha2** [comment]
FYI @JayavigneshwaranG - This is still an issue. General question/comment, instead of a simple find/replace of the string, what about a callback function for formatting?

`Highcharts` (which I am sure you are aware of) is a javascript charing library that enables this with great success. Here is an example in their docs:

https://api.highcharts.com/highcharts/tooltip.formatter

https://api.highcharts.com/class-reference/Highcharts#.TooltipFormatterCallbackFunction

------

Keeping the `format` property as it is now is fine, but I think a callback function that reveals much more about the context and allows for dynamic string generation could be a useful feature. This is not the request of this issue, but just something I thought I would bring up for discussion/consideration.

**Comment by dharanidharandharmasivam** [comment]
Hi @walsha2 ,

Thank for the information and your suggestions. We suspect that your requirement is to hide the y-axis (vertical axis) labels, so you have given labelFormat in primaryYAxis as an empty string. By default, the format applied to y-axis will be displayed the tooltip y value. If you need y-axis labels, axis line needs to be hidden, you can specify **primaryYAxis.isVisible** property to **false** and the tooltip values will be displayed properly. 

Also, you can change the format in the tooltip rendering time using the **onTooltipRender** event. Find the code snippet below to achieve this.

> SfCartesianChart(
        onTooltipRender: (TooltipArgs args) {
                final int pointIndex = args.pointIndex;
                args.header = 'Tooltip header';
                args.text = args.dataPoints[pointIndex].x.toString() +
                    ' has value of: ' +
                    args.dataPoints[pointIndex].y.toString();
              },
)

To know more about events, kindly find the chart [help document](https://help.syncfusion.com/flutter/chart/events).

Hope this helps. 

Thanks,
Dharani.

**Comment by walsha2** [comment]
@dharanidharandharmasivam oh wow, I did even know about the docs you linked to. Those are good! Maybe add them to the `README.md` and pub.dev link as well?

I have only been using the API reference

https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/charts-library.html

------

You are correct it works when `isVisible=false`. I want to hide the labels on the y axis, but still show the y-axis grid lines. Setting `isVisible=false` will remove the grid lines.

--------

I just confirmed that this works as you describe when using `onTooltipRender `. Will close this issue.
